src/main/java/io/kestra/plugin/azure/runner/AzureBatchScriptRunner.java

```diff
@@ -0,0 +1,183 @@
+
+            SharedAccess task = SharedAccess.builder()
+                .id(SharedAccess.class.getSimpleName())
+                .type(io.kestra.plugin.azure.storage.blob.List.class.getName())
+                .endpoint(this.endpoint)
+                .connectionString(blobStorage.getConnectionString())
+                .container(blobStorage.getContainerName())
+                .name(blobName)
+                .expirationDate("{{ now() | dateAdd(1, 'DAYS')  }}")
+                .permissions(Set.of(SharedAccess.Permission.READ))
+                .build();
+
+            SharedAccess.Output sas = task.run(runContext);
+
+            return ResourceFile.builder()
+                .filePath(file.startsWith("/") ? file.substring(1) : file)
+                // Use path to eventually deduplicate leading '/'
+                .httpUrl(sas.getUri().toString())
+                .build();
+        })).toList();
+
+        AbstractLogConsumer logConsumer = commandsWrapper.getLogConsumer();
+        Create createJob = Create.builder()
+            .id("create")
+            .type(Create.class.getName())
+            .account(this.account)
+            .accessKey(this.accessKey)
+            .endpoint(this.endpoint)
+            .poolId(this.poolId)
+            .maxDuration(this.waitUntilCompletion)
+            .job(
+                Job.builder()
+                    .id(jobId)
+                    .build()
+            )
+            .tasks(List.of(
+                Task.builder()
+                    .id("task-" + jobId)
+                    .constraints(
+                        TaskConstraints.builder()
+                            .maxWallClockTime(this.waitUntilCompletion)
+                            .maxTaskRetryCount(0)
+                            .build()
+                    )
+                    .interpreter(command.get(0))
+                    .interpreterArgs(command.size() > 1 ? new String[]{command.get(1)} : new String[0])
+                    .commands(command.size() > 2 ? command.subList(2, command.size()) : Collections.emptyList())
+                    .resourceFiles(resourceFiles)
+                    .outputFiles(filesToDownload)
+                    .containerSettings(
+                        TaskContainerSettings.builder()
+                            .workingDirectory(ContainerWorkingDirectory.TASK_WORKING_DIRECTORY)
+                            .registry(registry)
+                            .imageName(commandsWrapper.getContainerImage())
+                            .build()
+                    )
+                    .build()
+            ))
+            .logConsumer(new AbstractLogConsumer() {
+                @Override
+                public void accept(String log, Boolean isStdErr) {
+                    log.replaceAll("(?!^)(::\\{)", System.lineSeparator() + "::{")
+                        .lines()
+                        .forEach(line -> logConsumer.accept(line, isStdErr));
+                }
+            })
```

Comment:
You should not create a new one but use the one from the command.
Some tasks (like DBT) has special treatment for logs

Reply:
That's just an extra behaviour I add (see the logConsumer.accept which forwards the call to the commandsWrapper.logConsumer). I do this split behaviour because sometimes the logs even though they are multiple lines from commands, they are interpreted as a single line making the ::{"outputs"...}:: syntax not working